File: JavaScript/checkOutApp.js
// Check out app
const prompt = require("prompt-sync")();

function getStoreDetails() {
	return "\n\nSEMICOLON STORES \nMAIN BRANCH \nLOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS. \nTEL: 03938373320 \nDATE: " + new Date().toLocaleString();
}

function printCustomerInvoice(items, customer, cashier, discount) {
	console.log("Cashier: " + cashier);
	console.log("Customer Name: " + customer);
	console.log("=".repeat(60));
	console.log("\t\tITEM\tQUANTITY\tPRICE\tTOTAL(NGN)");
	console.log("-".repeat(60));

	let total = 0;
	for (let item of items) {
		let itemTotal = item.price * item.quantity;
		total += itemTotal;
		console.log("\t\t" + item.name + "\t" + item.quantity + "\t\t" + item.price + "\t" + itemTotal.toFixed(2));
	}

	console.log("-".repeat(60));

	let discountAmount = total * (discount / 100);
	let vat = total * 0.175;
	let billTotal = total - discountAmount + vat;

	console.log("\t\t\tSub Total:\t\tN" + total.toFixed(2));
	console.log("\t\t\tDiscount:\t\tN" + discountAmount.toFixed(2));
	console.log("\t\t\tVAT @ 17.50%:\t\tN" + vat.toFixed(2));
	console.log("=".repeat(60));
	console.log("\t\t\tBill Total:\t\tN" + billTotal.toFixed(2));
	console.log("=".repeat(60));
	console.log("THIS IS NOT A RECEIPT. KINDLY PAY N" + billTotal.toFixed(2));
	console.log("=".repeat(60));

	return billTotal;
}

function printCustomerReceipt(items, customer, cashier, discount, payment) {
	console.log("Cashier: " + cashier);
	console.log("Customer Name: " + customer);
	console.log("=".repeat(60));
	console.log("\t\tITEM\tQUANTITY\tPRICE\tTOTAL(NGN)");
	console.log("-".repeat(60));

	let total = 0;
	for (let item of items) {
		let itemTotal = item.price * item.quantity;
		total += itemTotal;
		console.log("\t\t" + item.name + "\t" + item.quantity + "\t\t" + item.price + "\t" + itemTotal.toFixed(2));
	}

	console.log("-".repeat(60));

	let discountAmount = total * (discount / 100);
	let vat = total * 0.175;
	let billTotal = total - discountAmount + vat;
	let balance = payment - billTotal;

	console.log("\t\t\tSub Total:\t\tN" + total.toFixed(2));
	console.log("\t\t\tDiscount:\t\tN" + discountAmount.toFixed(2));
	console.log("\t\t\tVAT @ 17.50%:\t\tN" + vat.toFixed(2));
	console.log("=".repeat(60));
	console.log("\t\t\tBill Total:\t\tN" + billTotal.toFixed(2));
	console.log("\t\t\tAmount Paid:\t\tN" + payment.toFixed(2));
	console.log("\t\t\tBalance:\t\tN" + balance.toFixed(2));
	console.log("=".repeat(60));
	console.log("\t\tTHANK YOU FOR YOUR PATRONAGE");
	console.log("=".repeat(60));
}

function main() {
	console.log("WELCOME TO SEMICOLON STORES\n");

	let customerName = prompt("What is the customer's Name? ");
	let items = [];
	let choice = "yes";

	while (choice.toLowerCase() === "yes") {
		let name = prompt("What did the customer buy? ");
		let price = parseFloat(prompt("How much per unit? "));
		if (price <= 0) {
			console.log("Enter a valid price");
			continue;
		}
		let quantity = parseInt(prompt("How many units? "), 10);

		items.push({ name, price, quantity });

		choice = prompt("Add more items? (yes/no): ");
		while (choice.toLowerCase() !== "yes" && choice.toLowerCase() !== "no") {
			choice = prompt("Please enter yes or no: ");
		}
	}

	let cashierName = prompt("What is your Name (cashier)? ");
	let discount = parseFloat(prompt("How much discount (%) will customer get? "));
	if (discount < 0 || discount > 100) {
		console.log("Discount must be between 0 and 100!");
		discount = 0;
	}

	console.log(getStoreDetails());
	let billTotal = printCustomerInvoice(items, customerName, cashierName, discount);

	let payment = parseFloat(prompt("How much did the customer give to you? "));
	if (payment <= 0) {
		console.log("Invalid or negative payment!");
		return;
	}

	console.log(getStoreDetails());
	printCustomerReceipt(items, customerName, cashierName, discount, payment);
}

main();

File: oldSnacks/Python/check_out_app.py
from datetime import datetime

class CheckOutAppFunctions:

	def get_store_details(self):
		return (f"\nSEMICOLON STORES \nMAIN BRANCH \nLOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS. \nTEL: 03938373320 \n{datetime.now()}")

	def print_customer_invoice(self, items, customer, cashier, discount):
		print(f"Cashier: {cashier}")
		print(f"Customer Name: {customer}")
		print("=" * 60)
		print("\t\tITEM\tQUANTITY\tPRICE\tTOTAL(NGN)")
		print("-" * 60)

		sub_total = 0
		for item in items:
			total = item['quantity'] * item['price']
			sub_total += total
			print(f"\t\t{item['name']}\t{item['quantity']}\t\t{item['price']}\t{total:.2f}")

		print("-" * 60)

		discount_amount = sub_total * (discount / 100)
		vat = sub_total * 0.175
		bill_total = sub_total - discount_amount + vat

		print(f"\n\t\t\tSub Total: \t\t{sub_total:.2f}")
		print(f"\t\t\tDiscount: \t\t{discount_amount:.2f}")
		print(f"\t\t\tVAT @ 17.50%: \t\t{vat:.2f}")
		print("=" * 60)
		print(f"\t\t\tBill Total: \t\t{bill_total:.2f}")
		print("=" * 60)
		print(f"THIS IS NOT A RECEIPT. KINDLY PAY {bill_total:.2f}")
		print("=" * 60)

	def print_customer_receipt(self, items, customer, cashier, discount, payment):
		print(f"Cashier: {cashier}")
		print(f"Customer Name: {customer}")
		print("=" * 60)
		print("\t\tITEM\tQUANTITY\tPRICE\tTOTAL(NGN)")
		print("-" * 60)

		sub_total = 0
		for item in items:
			total = item['quantity'] * item['price']
			sub_total += total
			print(f"\t\t{item['name']}\t{item['quantity']}\t\t{item['price']}\t{total:.2f}")

		print("-" * 60)

		discount_amount = sub_total * (discount / 100)
		vat = sub_total * 0.175
		bill_total = sub_total - discount_amount + vat
		balance = payment - bill_total

		print(f"\n\t\t\tSub Total: \t\t{sub_total:.2f}")
		print(f"\t\t\tDiscount: \t\t{discount_amount:.2f}")
		print(f"\t\t\tVAT @ 17.50%: \t\t{vat:.2f}")
		print("=" * 60)
		print(f"\t\t\tBill Total: \t\t{bill_total:.2f}")
		print(f"\t\t\tAmount Paid: \t\t{payment:.2f}")
		print(f"\t\t\tBalance: \t\t{balance:.2f}")
		
		if balance < 0:
			print(f"\n\t{customer}, please pay your balance of N{abs(balance):.2f}")

		print("=" * 60)
		print("\t\tTHANK YOU FOR YOUR PATRONAGE")
		print("=" * 60)


def main():
	checkout = CheckOutAppFunctions()
	
	customer_name = input("What is the customer's Name? ")
	
	items = []
	choice = "yes"

	while choice.lower() == "yes":
		item_bought = input("What did the customer buy? ")

		price_per_unit = float(input("How much per unit? "))
		if price_per_unit <= 0:
			print("Invalid price input")
			continue

		quantity = int(input("How many units? "))
		if quantity <= 0:
			print("Invalid number of units.")
			continue

		items.append({
			'name': item_bought,
			'price': price_per_unit,
			'quantity': quantity
		})

		choice = input("Add more items? (yes or no) ").lower()

		while choice not in ["yes", "no"]:
			choice = input("Add more items? ").lower()

	cashier_name = input("Enter cashier's name: ")

	discount = float(input("How much discount will customer get? "))
	if discount < 0 or discount > 100:
		print("Discount must be between 0 and 100!")

	print(checkout.get_store_details())
	checkout.print_customer_invoice(items, customer_name, cashier_name, discount)

	payment = float(input("How much did the customer give to you? "))
	if payment <= 0:
		print("Invalid or negative payment!")

	print(checkout.get_store_details())
	checkout.print_customer_receipt(items, customer_name, cashier_name, discount, payment)


if __name__ == "__main__":
	main()
